src/boosting/gbdt.cpp

```diff
@@ -547,15 +547,18 @@ void GBDT::SaveModelToFile(int num_iteration, const char* filename) const {
     output_file << pairs[i].second << "=" << std::to_string(pairs[i].first) << std::endl;
   }
 
-  output_file << std::endl << "feature information:" << std::endl;
-  for (size_t i = 0; i < max_feature_idx_ + 1; ++i) {
-    output_file << feature_names_[i] << "=" << feature_infos_[i] << std::endl;
+  if (feature_infos_.size() == feature_names_.size()) {
+      output_file << std::endl << "feature information:" << std::endl;
+      for (size_t i = 0; i < max_feature_idx_ + 1; ++i) {
+        output_file << feature_names_[i] << "=" << feature_infos_[i] << std::endl;
+      }
```

Comment:
and another question, shall `SaveModelToFile()` be split into two functions, one that saves a string, and the other one that saves the string to a file? Because it seems inconsistent to me that we have `LoadModelFromString()` but there is no `SaveModelToString()`. I can take care of that.